src/components/newsletter-form.js

```diff
@@ -34,14 +34,28 @@ const StyledInput = styled(TextField)(newsletterInputStyles);
 export function useNewsletterForm() {
   const [loading, setLoading] = useState(false);
   const [success, setSuccess] = useState(false);
+  const [error, setError] = useState(null);
 
   return {
     loading,
     success,
+    error,
     async handleSubmit(event) {
       event.preventDefault();
       const email = event.target.email.value;
 
+      const emailRegex = new RegExp(
+        /^[A-Za-z0-9_!#$%&'*+/=?`{|}~^.-]+@[A-Za-z0-9.-]+$/,
+        'gm'
```

Comment:
Since this regex is checking for the start and end of the string (`^` and `$`), we don't need these global and multiline flags here. They would only apply if we're looking for multiple matches in the same string or if we're looking for matches on multiple lines of a string, neither of which apply here.

So this regex can just be expressed in it's literal form, `/pattern/.test(string)`